roomserver/internal/perform/perform_admin.go

```diff
@@ -231,3 +232,37 @@ func (r *Admin) PerformAdminEvacuateUser(
 	}
 	return nil
 }
+
+func (r *Admin) PerformAdminPurgeRoom(
+	ctx context.Context,
+	req *api.PerformAdminPurgeRoomRequest,
+	res *api.PerformAdminPurgeRoomResponse,
+) error {
+	if _, _, err := gomatrixserverlib.SplitID('!', req.RoomID); err != nil {
```

Comment:
Would be nice to comment this. It's just basically checking that the input starts with `!`, but it's a nice validation check on the off chance this API is abused.